qa/rpc-tests/nov152018_forkactivation.py

```diff
@@ -0,0 +1,277 @@
+            old_sha256 = block.sha256
+            block.hashMerkleRoot = block.calc_merkle_root()
+            block.solve()
+            # Update the internal state just like in next_block
+            self.tip = block
+            if block.sha256 != old_sha256:
+                self.block_heights[block.sha256] = self.block_heights[old_sha256]
+                del self.block_heights[old_sha256]
+            self.blocks[block_number] = block
+            return block
+
+        # shorthand for functions
+        block = self.next_block
+
+        # Create a new block
+        block(0)
+        save_spendable_output()
+        yield accepted()
+
+        # Now we need that block to mature so we can spend the coinbase.
+        test = TestInstance(sync_every_block=False)
+        for i in range(99):
+            block(5000 + i)
+            test.blocks_and_transactions.append([self.tip, True])
+            save_spendable_output()
+        yield test
+
+        # collect spendable outputs now to avoid cluttering the code later on
+        out = []
+        for i in range(100):
+            out.append(get_spendable_output())
+
+        # Let's build some blocks and test them.
+        for i in range(15):
+            n = i + 1
+            block(n)
+            yield accepted()
+
+        # Start moving MTP forward
+        bfork = block(5555)
+        bfork.nTime = NOV152018_START_TIME - 1
+        update_block(5555)
+        yield accepted()
+
+        # Get to one block of the Nov 15, 2018 HF activation
+        for i in range(5):
+            block(5100 + i)
+            test.blocks_and_transactions.append([self.tip, True])
+        yield test
+
+        # Check that the MTP is just before the configured fork point.
+        assert_equal(node.getblockheader(node.getbestblockhash())['mediantime'],
+                     NOV152018_START_TIME - 1)
+
+        # Check that block with small transactions, non push only signatures and
+        # non clean stack are still accepted.
+        small_tx_block = block(
+            4444, out[0], MIN_TX_SIZE - 1, pushonly=False, cleanstack=False)
+        assert_equal(len(small_tx_block.vtx[1].serialize()), MIN_TX_SIZE - 1)
+        yield accepted()
+
+        # Now MTP is exactly the fork time. Small transaction are now rejected.
+        assert_equal(node.getblockheader(node.getbestblockhash())['mediantime'],
+                     NOV152018_START_TIME)
+
+        # Now that the fork activated, it is not possible to have
+        # small transactions anymore.
+        bestblockhash = self.nodes[0].getbestblockhash()
+        small_tx_block = block(4445, out[1], MIN_TX_SIZE - 1)
+        assert_equal(len(small_tx_block.vtx[1].serialize()), MIN_TX_SIZE - 1)
+        #check that this block was rejected.
+        yield rejected()
+        #yield rejected(RejectResult(16, b'bad-txns-undersize'))
+        assert_equal(bestblockhash, self.nodes[0].getbestblockhash())
```

Comment:
Shouldn't this test the boundary - as in also make sure that `MIN_TX_SIZE` just barely goes through?